gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1496,6 +1495,8 @@ private String listStorageObjectsAndPrefixesPage(
       listObject.setPrefix(objectNamePrefix);
     }
 
+    listObject = initializeRequest(listObject, bucketName);
```

Comment:
What information `initializeRequest` needs that we can not call it at the very beginning?

Reply:
The set prefix call at L:1495. Without setting prefix, the request looks like a list at bucket root level.

Reply:
Can we inline it in the `initializeRequest` then?

```java
Storage.Objects.List listObject =
        initializeRequest(
                storage.objects().list(bucketName).setPrefix(emptyToNull(objectNamePrefix)),
                bucketName);
```